From 2a73966cb8cbed973bb83a8ed8c6237f91fefe8a Mon Sep 17 00:00:00 2001
From: atkio <bot@example.com>
Date: Sat, 27 Jun 2026 23:20:41 +0900
Subject: [PATCH] Increase ESP32 i386 TLB size

---
 i386.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/i386.c b/i386.c
index db323f5..7c3ad3f 100644
--- a/i386.c
+++ b/i386.c
@@ -511,7 +511,7 @@ static inline int get_IOPL(CPUI386 *cpu)
 #define CR0_PG (1<<31)
 #define CR0_WP (0x10000)
 #ifdef BUILD_ESP32
-#define tlb_size 256
+#define tlb_size 1024
 #else
 #define tlb_size 512
 #endif
-- 
2.50.0.windows.1

