From 31aac109d209b747f0bdda723a0b3b5b106e9653 Mon Sep 17 00:00:00 2001
From: atkio <bot@example.com>
Date: Sat, 27 Jun 2026 23:20:27 +0900
Subject: [PATCH] Place ESP32 OPL tables in external BSS

---
 emu8950.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/emu8950.c b/emu8950.c
index 33334c4..6380e3e 100644
--- a/emu8950.c
+++ b/emu8950.c
@@ -9,6 +9,13 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef BUILD_ESP32
+#include "esp_attr.h"
+#define EMU8950_LARGE_BSS EXT_RAM_BSS_ATTR
+#else
+#define EMU8950_LARGE_BSS
+#endif
+
 #ifndef INLINE
 #if defined(_MSC_VER)
 #define INLINE __inline
@@ -90,7 +97,7 @@ static uint16_t logsin_table[PG_WIDTH / 4] = {
 };
 /* clang-format on */
 
-static uint16_t wave_table_map[4][PG_WIDTH];
+static EMU8950_LARGE_BSS uint16_t wave_table_map[4][PG_WIDTH];
 
 /* pitch modulator */
 #define PM_PG_BITS 3
@@ -151,8 +158,8 @@ static double kl_table[16] = {dB2(0.000),  dB2(9.000),  dB2(12.000), dB2(13.875)
                               dB2(16.875), dB2(17.625), dB2(18.000), dB2(18.750), dB2(19.125), dB2(19.500),
                               dB2(19.875), dB2(20.250), dB2(20.625), dB2(21.000)};
 
-static uint32_t tll_table[8 * 16][1 << TL_BITS][4];
-static int32_t rks_table[2][32][2];
+static EMU8950_LARGE_BSS uint32_t tll_table[8 * 16][1 << TL_BITS][4];
+static EMU8950_LARGE_BSS int32_t rks_table[2][32][2];
 
 #define min(i, j) (((i) < (j)) ? (i) : (j))
 #define max(i, j) (((i) > (j)) ? (i) : (j))
@@ -1352,4 +1359,4 @@ void OPL_writeADPCMData(OPL *opl, uint8_t type, uint32_t start, uint32_t length,
       OPL_ADPCM_writeROM(opl->adpcm, start, length, data);
     }
   }
-}
\ No newline at end of file
+}
-- 
2.50.0.windows.1

